### PR TITLE
Drop presto-orc's dependency on ConnectorSession

### PR DESCRIPTION
**Pull request by NikhilCollooru**
depended by https://github.com/facebookexternal/presto-facebook/pull/955

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

**Comment by NikhilCollooru** [comment]
> This is on the right direction. Unfortunately there are a lot of places to be taken care of in order to migrate `ConnectorSession` to `SqlFunctionProperties`. One big package is `presto-main/src/main/java/com/facebook/presto/operator/scalar`. We will need to migrate most of them to use `SqlFunctionProperties`
> 
> ```
> ➜  presto git:(14375) grep -r ConnectorSession presto-main/src/main/java/com/facebook/presto/operator/scalar | sed 's/.java:.*//' | sort | uniq -c
>    2 presto-main/src/main/java/com/facebook/presto/operator/scalar/ArrayJoin
>    3 presto-main/src/main/java/com/facebook/presto/operator/scalar/BuiltInScalarFunctionImplementation
>   41 presto-main/src/main/java/com/facebook/presto/operator/scalar/DateTimeFunctions
>    2 presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonFunctions
>    3 presto-main/src/main/java/com/facebook/presto/operator/scalar/JsonOperators
>    2 presto-main/src/main/java/com/facebook/presto/operator/scalar/MapFromEntriesFunction
>    2 presto-main/src/main/java/com/facebook/presto/operator/scalar/MapToMapCast
>    3 presto-main/src/main/java/com/facebook/presto/operator/scalar/SequenceFunction
>    2 presto-main/src/main/java/com/facebook/presto/operator/scalar/SessionFunctions
>   13 presto-main/src/main/java/com/facebook/presto/operator/scalar/annotations/ParametricScalarImplementation
> ```

I removed the usages in those files. The grep is probably searching the comments as well. I will change that . But i think the tests are failing because I removed ConnectorSession from castTotimestamp() function in DateOperators. How to resolve that error ?

**Comment by highker** [comment]
hmmm.... `DateTimeFunctions` still has dependency on `ConnectorSession`, doesn't it?

**Comment by NikhilCollooru** [comment]
> hmmm.... `DateTimeFunctions` still has dependency on `ConnectorSession`, doesn't it?

Yes. sorry didnt check correctly. False alert.

Also for ```session.getStartTime()``` there is no equivalent in ```SqlFunctionProperties```. How to handle that ?

**Comment by highker** [comment]
Feel free to refactor `SqlFunctionProperties` to include `startTime`

File: presto-hive/src/main/java/com/facebook/presto/hive/FilteringPageSource.java [hunk 1]
```diff
@@ -119,7 +119,7 @@ public FilteringPageSource(
         }
         else {
             RowExpression expression = replaceExpression(optimizedRemainingPredicate, variableToInput);
-            this.filterFunction = new FilterFunction(session,
+            this.filterFunction = new FilterFunction(session.getSqlFunctionProperties(),
```
Comment (highker) [inline presto-hive/src/main/java/com/facebook/presto/hive/FilteringPageSource.java:122]: move `session.getSqlFunctionProperties()` to its own line

File: presto-main/src/main/java/com/facebook/presto/operator/scalar/BuiltInScalarFunctionImplementation.java [hunk 2]
```diff
@@ -131,13 +131,13 @@ public ScalarImplementationChoice(
 
             List<Class<?>> parameterList = methodHandle.type().parameterList();
             boolean hasSession = false;
```
Comment (highker) [inline presto-main/src/main/java/com/facebook/presto/operator/scalar/BuiltInScalarFunctionImplementation.java:133]: `hasProperties`; same for the stack var

File: presto-spi/src/main/java/com/facebook/presto/spi/function/SqlFunctionProperties.java [hunk 3]
```diff
@@ -26,19 +27,28 @@
...
+            String sessionUser)
     {
         this.parseDecimalLiteralAsDouble = parseDecimalLiteralAsDouble;
         this.legacyRowFieldOrdinalAccessEnabled = legacyRowFieldOrdinalAccessEnabled;
         this.timeZoneKey = requireNonNull(timeZoneKey, "timeZoneKey is null");
         this.legacyTimestamp = legacyTimestamp;
         this.legacyMapSubscript = legacyMapSubscript;
+        this.sessionStartTime = sessionStartTime;
+        this.sessionLocale = sessionLocale;
+        this.sessionUser = sessionUser;
```
Comment (highker) [inline presto-spi/src/main/java/com/facebook/presto/spi/function/SqlFunctionProperties.java:51]: requireNonNull

**Pull request merged**